crates/core/src/config/mod.rs

```diff
@@ -108,25 +121,19 @@ impl Config {
 
     /// Returns a `Option<PathBuf>` to the directory which will hold the
     /// temporary zip archives.
-    /// This will return `None` if no `wow_directory` is set in the config.
+    /// This will return `None` if flavor does not have a directory.
     pub fn get_download_directory_for_flavor(&self, flavor: Flavor) -> Option<PathBuf> {
-        match self.get_addon_directory_for_flavor(&flavor) {
-            Some(dir) => {
-                // The path to the directory which hold the temporary zip archives
-                let dir = dir.parent().expect("Expected Addons folder has a parent.");
-                Some(dir.to_path_buf())
-            }
-            None => None,
-        }
+        self.wow.directories.get(&flavor).cloned()
```

Comment:
Before we used the `Interface` folder, but this is using the `\_flavor\_` directory now?

Reply:
I just thought this was easier, honestly.
I am happy to change it if you think it makes the most sense?

Reply:
Probably doesn't matter either way as we clean up those temp files. Your call.